src/test/java/com/pmarko09/school_supervision/service/StudentServiceTest.java

```diff
@@ -0,0 +1,286 @@
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.when;
+
+public class StudentServiceTest {
+
+    StudentService studentService;
+    StudentRepository studentRepository;
+    SchoolClassRepository schoolClassRepository;
+    SubjectRepository subjectRepository;
+    ExamResultRepository examResultRepository;
+    StudentMapper studentMapper;
+
+    @BeforeEach
+    void setup() {
+        this.studentRepository = Mockito.mock(StudentRepository.class);
+        this.schoolClassRepository = Mockito.mock(SchoolClassRepository.class);
+        this.subjectRepository = Mockito.mock(SubjectRepository.class);
+        this.examResultRepository = Mockito.mock(ExamResultRepository.class);
+        this.studentMapper = Mappers.getMapper(StudentMapper.class);
+        this.studentService = new StudentService(studentRepository, schoolClassRepository,
+                subjectRepository, examResultRepository, studentMapper);
+    }
+
+    @Test
+    void getAllStudents_CorrectData_StudentsDtoReturned() {
+
+        //given
+        Student student = new Student();
+        student.setId(1L);
+        student.setFirstname("Jan");
+        student.setLastname("Bebb");
+        student.setEmail("aa@");
+        student.setCardNumber(123);
+
+        when(studentRepository.findAll()).thenReturn(List.of(student));
+        StudentDTO studentDTO = studentMapper.toDto(student);
+
+        //when
+        Set<StudentDTO> result = studentService.getStudents();
+
+        //then
+        assertEquals(1, result.size());
+        assertTrue(result.contains(studentDTO));
+    }
+
+    @Test
+    void getStudent_DataCorrect_StudenttDtoReturned() {
+        //given
+        Student student = new Student();
+        student.setId(1L);
+        student.setFirstname("Jan");
+        student.setLastname("Bebb");
+        student.setEmail("aa@");
+        student.setCardNumber(123);
+
+        when(studentRepository.findById(1L)).thenReturn(Optional.of(student));
+        studentMapper.toDto(student);
+
+        //when
+        StudentDTO result = studentService.getStudent(1L);
+
+        //then
+        assertEquals("Jan", result.getFirstname());
+        assertEquals("Bebb", result.getLastname());
+        assertEquals("aa@", result.getEmail());
+        assertEquals(123, result.getCardNumber());
+    }
+
+    @Test
+    void addStudent_DataCorrect_StudentDtoReturned() {
+        //given
+        Student student = new Student();
+        student.setId(1L);
+        student.setFirstname("Jan");
+        student.setLastname("Bebb");
+        student.setEmail("aa@");
+        student.setCardNumber(123);
+
+        StudentValidation.validateStudentData(student);
```

Comment:
ten test powinien sprawdzac dodanie studenta dlaczego mamy tutaj wywolana walidację -> jesli chcemy  przetestowac walidacje to powinien to byc dla niej oddzielny test